pkg/image/validation/validation.go

```diff
@@ -23,7 +23,10 @@ func ValidateDefinition(ctx *image.Context) []FailedValidation {
 	}
 	for _, v := range validations {
 		componentFailures := v(ctx)
-		failures = append(failures, componentFailures...)
+
+		if len(componentFailures) > 0 {
+			failures[componentFailures[0].Component] = componentFailures
```

Comment:
This is not what I imagined. I thought that `validations` would map a section / component to its validation function and use that key in order to store the failures. This way you don't have to set `Component` for each object and missing it will not lead into strange behaviour.

Reply:
I don't think it makes sense for each `validate*` method to return a map though. Each will only ever validate things in one component, so it feels weird to return a map with a single key from each of those methods.